modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/MessageEnricherObjectFactory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.config.spring.dsl.processor;
+
+
+import org.mule.runtime.config.spring.dsl.api.ObjectFactory;
+import org.mule.runtime.core.api.AnnotatedObject;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.construct.FlowConstructAware;
+import org.mule.runtime.core.api.context.MuleContextAware;
+import org.mule.runtime.core.api.processor.MessageProcessor;
+import org.mule.runtime.core.enricher.MessageEnricher;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+
+/**
+ * {@link org.mule.runtime.config.spring.dsl.api.ObjectFactory} to create a {@link org.mule.runtime.core.enricher.MessageEnricher}.
+ */
+public class MessageEnricherObjectFactory implements ObjectFactory<MessageEnricher>, MuleContextAware, FlowConstructAware, AnnotatedObject
+{
+
+    private MessageProcessor messageProcessor;
+    private String source;
+    private String target;
+    private List<MessageEnricher.EnrichExpressionPair> enrichExpressionPairs = new ArrayList<>();
+    private FlowConstruct flowConstruct;
+    private MuleContext muleContext;
+    private Map<QName, Object> annotations;
```

Comment:
We are going to get rid of this eventually so extensions do not need to declare Annotations. That will be possible with this new approach.